std/assembly/string.ts

```diff
@@ -637,23 +637,25 @@ export namespace String {
     export function byteLength(str: string, nullTerminated: bool = false): i32 {
       var strOff = changetype<usize>(str);
       var strEnd = strOff + <usize>changetype<BLOCK>(changetype<usize>(str) - BLOCK_OVERHEAD).rtSize;
-      var bufLen = nullTerminated ? 1 : 0;
+      var bufLen = <i32>nullTerminated;
       while (strOff < strEnd) {
         let c1 = <u32>load<u16>(strOff);
         if (c1 < 128) {
-          if (nullTerminated && !c1) break;
-          bufLen += 1; strOff += 2;
+          // @ts-ignore: cast
+          if (nullTerminated & !c1) break;
+          bufLen += 1;
         } else if (c1 < 2048) {
-          bufLen += 2; strOff += 2;
+          bufLen += 2;
         } else {
           if ((c1 & 0xFC00) == 0xD800 && strOff + 2 < strEnd) {
             if ((<u32>load<u16>(strOff, 2) & 0xFC00) == 0xDC00) {
-              strOff += 4; bufLen += 4;
+              bufLen += 4; strOff += 4;
               continue;
             }
           }
-          strOff += 2; bufLen += 3;
+          bufLen += 3;
         }
+        strOff += 2;
```

Comment:
This was refactored in `byteLength` but not in `encode` - that intended?

Reply:
Yeah, `byteLength` also slightly refactored. I'll update PR's description

Reply:
What I meant is that the exact same redundant `strOff += 2` is used in `encode`, but there it's not refactored to outside the loop. Seems to me that this would make sense there as well?

Reply:
Right, done!